types.go

```diff
@@ -48,11 +48,25 @@ type TimerSubtypes struct {
 }
 
 // Runnable is a long running function intended to be launched in a goroutine.
-type Runnable func(ctx context.Context)
+type Runnable func(context.Context)
 
 // Runner exposes a Runnable through an interface
 type Runner interface {
-	Run(ctx context.Context)
+	Run(context.Context)
+}
+
+type MetricsRunner interface {
+	RunMetricsContext(context.Context)
+}
+
+func MaybeAppendRunnable(runnables []Runnable, maybeRunner interface{}) []Runnable {
```

Comment:
Single helper func to ensure we always start both the main work loop and the metrics loop too.